app/views/play/level/LevelPlaybackView.coffee

```diff
@@ -219,21 +222,27 @@ module.exports = class LevelPlaybackView extends CocoView
       @timePopup.show()
 
   onProgressTapStart: (e, touchData) ->
-    return unless application.isIPadApp
+    return unless application.isIPadApp or utils.isMobile()
     @onProgressEnter e
-    screenOffsetX = e.clientX ? touchData?.position.x ? 0
+    if e.handleObj.type == 'tapstart'
+      screenOffsetX = e.clientX ? touchData?.position.x ? 0
+    else
+      screenOffsetX = e.originalEvent.touches[0].clientX
```

Comment:
will any of these fields ever be null or empty causing undefined bug?

Reply:
ideally only  `tapstart` and `touchstart` hit his function so if not tapstart then should be touchstart so nothing would be undefined. but... who knows, let's add checking for safe usage.